packages/internal/src/lib-chainStorage.js

```diff
@@ -222,13 +222,15 @@ harden(makeStorageNodeChild);
 /**
  *
  * @param {import('@endo/far').ERef<StorageNode>} storageNode
- * @param {import('@endo/far').ERef<Marshaller>} marshaller
+ * @param {import('@endo/far').ERef<Marshaller & {
+ *   serializeAndStringify: (value: unknown) => string
+ * }>} marshaller
  * @returns {(value: unknown) => Promise<void>}
  */
 export const makeSerializeToStorage = (storageNode, marshaller) => {
   return async value => {
-    const marshalled = await E(marshaller).serialize(value);
-    const serialized = JSON.stringify(marshalled);
-    return E(storageNode).setValue(serialized);
+    const serializedP = E(marshaller).serializeAndStringify(value);
+    // @ts-expect-error M.callWhen
+    return E(storageNode).setValue(serializedP);
```

Comment:
Shouldn't this include generalization of StorageNode `setValue` to include `await`ing its argument?
```suggestion
    return E(storageNode).setValue(serializedP);
```

Reply:
IIUC, `M.callWhen` ensures the promise is resolved before calling the `setValue` method.

Reply:
Right, but I don't see the expected update to introduce that `M.callWhen`.

Reply:
update? isn't it already there?

https://github.com/Agoric/agoric-sdk/blob/2b027a4015687161ac3196fa39225f0697a65efc/packages/internal/src/lib-chainStorage.js#L39

Reply:
Hmm, then what's the TS error?

Reply:
The TS error is because the caller passes a `Promise<string>` and the callee is declared to accept `string`. `tsc` isn't aware of `M.callWhen` fixing things up in the middle. Hence `@tsc-expect-error M.callWhen`.

Reply:
I wonder if it wouldn't make more sense to deal with that at the source in lib-chainStorage.js by fibbing like `/** @type {(value: ERef<string>) => Promise<void>} */`, but if not then maybe provide more context here?
```suggestion
    // @ts-expect-error M.callWhen auto-awaits serializedP for setValue(value: string)
    return E(storageNode).setValue(serializedP);
```
But I get the shorthand now so these are completely non-blocking.

Reply:
fibbing means the `setValue` implementation won't be typechecked correctly. But I suppose it would make sense to put 1 `@ts-expect-error` or the like there rather than 1 for every caller.

Reply:
Reviewing [@endo/patterns](https://github.com/endojs/endo/blob/master/packages/patterns/README.md), it also looks like `ChainStorageNodeI` isn't quite right... I think it needs the following change:
```diff
 const ChainStorageNodeI = M.interface('StorageNode', {
-  setValue: M.callWhen(M.string()).returns(),
+  setValue: M.callWhen(M.await(M.string())).returns(),
   getPath: M.call().returns(M.string()),
   getStoreKey: M.callWhen().returns(M.record()),
   makeChildNode: M.call(M.string())
     .optional(M.splitRecord({}, { sequence: M.boolean() }, {}))
     .returns(M.remotable('StorageNode')),
 });
```